proto/substrait/type.proto

```diff
@@ -164,13 +164,15 @@ message Type {
   }
 
   message PrecisionTimestamp {
+    // Sub-second precision, 0 means the value given is in seconds, 3 is milliseconds, 6 microsecods, 9 is nanoseconds
     // Defaults to 6
     int32 precision = 1;
```

Comment:
Not related to these changes, but how does `Defaults to 6` work here given that default value if precision is not set will be 0, and we can't distinguish between
* 0 was set to indicate seconds.
* 0 was not set intentionally

Should we remove this entirely?
@westonpace

Reply:
Ah I did mean to remove that, as I don't think it works, like you say - https://github.com/substrait-io/substrait/commit/666aa60e7a9709283a62747c6c61667321cd80fc. Hope that's better!

(The commit did remove your approvals though @vbarua @westonpace )

Reply:
Good catch.  I agree that a "default" can't exist here.  The default of 6 is really "if you see a `Timestamp` or `TimestampTz` type then you can treat it as a `PrecisionTimestamp` or `PrecisionTimestampTz` type where the `precision = 6`.